datalad/local/export_archive.py

```diff
@@ -115,18 +110,20 @@ def _filter_tarinfo(ti):
                 compression) if archivetype == 'tar' else '')
 
         default_filename = "datalad_{.id}".format(dataset)
+        if filename is not None:
+            filename = Path(filename)
         if filename is None:
-            filename = default_filename  # in current directory
-        elif path.exists(filename) and path.isdir(filename):
-            filename = path.join(filename, default_filename) # under given directory
-        if not filename.endswith(file_extension):
-            filename += file_extension
+            filename = Path(default_filename)  # in current directory
+        elif filename.exists() and filename.is_dir():
+            filename = filename / default_filename # under given directory
+        if not filename.suffix == file_extension:
+            filename = filename.with_suffix(file_extension)
 
         root = dataset.path
-        # use dir inside matching the output filename
+        # use dir inside matching the output filename without suffix(es)
         # TODO: could be an option to the export plugin allowing empty value
         # for no leading dir
-        leading_dir = file_basename(filename)
+        leading_dir = filename.name[: -len(''.join(filename.suffixes))]
```

Comment:
I think it would be best to leave using that helper, because it is well tested and does operate differently and IMHO "more correctly" in current state of "extensions"

<details>
<summary>arguments -- try on "1_R1.1.1.tar.gz"</summary> 

```shell
(git)lena:~datalad/datalad-master[bf-push-github]git
$> python -c "from pathlib import Path; print(Path('1_R1.1.1.tar.gz').suffixes)"
['.1', '.1', '.tar', '.gz']
(dev3) 1 29947 [1].....................................:Tue 07 Jun 2022 11:15:38 AM EDT:.
(git)lena:~datalad/datalad-master[bf-push-github]git
$> git grep file_basename                                                       
datalad/local/add_archive_content.py:    file_basename,
datalad/local/add_archive_content.py:        archive_basename = file_basename(archive)
datalad/local/add_archive_content.py:                        fn_base, fn_ext = file_basename(fn, return_ext=True)
datalad/local/export_archive.py:        from datalad.utils import file_basename
datalad/local/export_archive.py:        leading_dir = file_basename(filename)
datalad/tests/test_utils.py:    file_basename,
datalad/tests/test_utils.py:def test_file_basename():
datalad/tests/test_utils.py:    eq_(file_basename('1'), '1')
datalad/tests/test_utils.py:    eq_(file_basename('d1/1'), '1')
datalad/tests/test_utils.py:    eq_(file_basename('/d1/1'), '1')
datalad/tests/test_utils.py:    eq_(file_basename('1.'), '1.')
datalad/tests/test_utils.py:    eq_(file_basename('1.tar.gz'), '1')
datalad/tests/test_utils.py:    eq_(file_basename('1.Tar.gz'), '1')
datalad/tests/test_utils.py:    eq_(file_basename('1._bak.gz'), '1')
datalad/tests/test_utils.py:    eq_(file_basename('1.tar.gz', return_ext=True), ('1', 'tar.gz'))
datalad/tests/test_utils.py:    eq_(file_basename('/tmp/1.tar.gz'), '1')
datalad/tests/test_utils.py:    eq_(file_basename('/tmp/1.longish.gz'), '1.longish')
datalad/tests/test_utils.py:    eq_(file_basename('1_R1.1.1.tar.gz'), '1_R1.1.1')
datalad/tests/test_utils.py:    eq_(file_basename('ds202_R1.1.1.tgz'), 'ds202_R1.1.1')
datalad/utils.py:def file_basename(name, return_ext=False):
datalad/utils.py:    dirname = file_basename(name)

```
</details>

hence
```suggestion
        leading_dir = file_basename(filename)
```

<details>
<summary>on a quick check -- no `str` is needed</summary> 

```shell
$> python -c "from pathlib import Path; from datalad.utils import file_basename; print(file_basename(Path('1_R1.1.1.tar.gz')))"
1_R1.1.1
```
</details>

Reply:
sure, if you say so